src/Compilers/CSharp/Portable/Symbols/FunctionPointerTypeSymbol.cs

```diff
@@ -0,0 +1,264 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Reflection;
+using Microsoft.Cci;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal class FunctionPointerTypeSymbol : TypeSymbol
+    {
+
+        public static FunctionPointerTypeSymbol CreateFunctionPointerTypeSymbolFromSource(FunctionPointerTypeSyntax syntax, Binder typeBinder, DiagnosticBag diagnostics, ConsList<TypeSymbol> basesBeingResolved)
+        {
+            var symbol = new FunctionPointerTypeSymbol();
+            var signature = FunctionPointerMethodSymbol.CreateMethodFromSource(symbol, syntax, typeBinder, diagnostics, basesBeingResolved);
+
+            symbol.Signature = signature;
+
+            return symbol;
+        }
+
+        public static CallingConvention GetCallingConvention(string convention) =>
+            convention switch
+            {
+                "" => CallingConvention.Default,
+                "cdecl" => CallingConvention.C,
+                "managed" => CallingConvention.Default,
+                "thiscall" => CallingConvention.ThisCall,
+                "stdcall" => CallingConvention.Standard,
+                _ => CallingConvention.Invalid,
+            };
+
+        private FunctionPointerTypeSymbol()
+        {
+        }
+
+        public MethodSymbol Signature { get; private set; } = null!;
```

Comment:
>public MethodSymbol Signature { get; private set; } = null!; [](start = 8, length = 60)

I think this property should be read-only. It should not be possible to update the value once set. Constructor should be able to do this without problems, I think.